### DISCUSSION TITLE
How does the aiohttp server asynchronous behaviour works?

### DISCUSSION THREAD

**Discussion by miguelvieiraramos** [aio-libs/aiohttp#7076] [Q&A] [answered]
Hello, everyone :)

I couldn't find in the documentation what it actually means to be a asynchronous server. 
Does it means that aiohttp server works like nodejs? Meaning that is a single-threaded instance that when one request is awaiting the other request takes control.
Other possibility would be to work like flask that each instance can handle one request at a time, but when using [async](https://flask.palletsprojects.com/en/2.2.x/async-await/#performance) it means that you can use async features inside the request but the instance still can handle only one request at a time.

**Comment by Dreamsorcerer** [comment, marked as answer]
Yes, the former (handling multiple requests in a single thread). More generally, it is a server implemented natively in asyncio, so take a look at how asyncio behaves.
